controllers/ingressnodefirewallnodestate_controller_test.go

```diff
@@ -0,0 +1,106 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	infv1alpha1 "github.com/openshift/ingress-node-firewall/api/v1alpha1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ingressNodeFirewallRules map[string][]infv1alpha1.IngressNodeFirewallRules
+
+// ebpfSingletonMock implements ebpfDaemon.
+type ebpfSingletonMock struct{}
+
+func (e *ebpfSingletonMock) SyncInterfaceIngressRules(
+	ifaceIngressRules map[string][]infv1alpha1.IngressNodeFirewallRules, isDelete bool) error {
+	ingressNodeFirewallRules = ifaceIngressRules
+	return nil
+}
+
+var _ = Describe("IngressNodeFirewallNodeState controller", func() {
+	const (
+		timeout  = time.Second * 10
+		interval = time.Millisecond * 250
+	)
+
+	// Mock the ebpfDaemon.
+	mock = &ebpfSingletonMock{}
+	ctx := context.TODO()
+
+	BeforeEach(func() {
+		By(fmt.Sprintf("By creating a new Node object with name %s", daemonReconcilerNodeName))
```

Comment:
nit: no need to add "By" to your string this comment for all your new test cases

Reply:
done

Reply:
done